Build fails with malformed ldflags when git tags are missing locally

<!-- Please use this template while reporting a bug and provide as much info as possible. Thanks!-->
**What happened:**
When attempting to build KubeEdge locally using `make all` in a repository where git tags have not been fetched (e.g., a shallow clone or a direct ZIP download), the build script (`hack/lib/golang.sh`) outputs a warning message `"⚠️ No git tags found."` to `stdout`. 

Because this warning is outputted to `stdout`, it gets captured by the `$(kubeedge::version::ldflags)` execution and injected directly into the `GOLDFLAGS` variable. This causes the Go compiler to crash with malformed linker arguments:

```text
building github.com/kubeedge/kubeedge/cloud/cmd/cloudcore
+ go build -o _output/local/bin/cloudcore -gcflags= -ldflags '-s -w -buildid= 
⚠️  No git tags found.
⚠️  Falling back to default version: v0.0.0
⚠️  To avoid this, consider running: git fetch --tags
-X github.com/kubeedge/kubeedge/pkg/version.buildDate=...
# github.com/kubeedge/kubeedge/cloud/cmd/cloudcore
usage: link [options] main.o
...
make: *** [all] Error 1
```
______________________________________



**What you expected to happen**: The build should fall back to v0.0.0 gracefully and continue compiling without breaking the Go linker. The warning messages in hack/lib/golang.sh should be redirected to stderr (>&2) so they are not captured by the ldflags variable.

**How to reproduce it (as minimally and precisely as possible)**:
- Clone the repository without tags: git clone --depth 1 https://github.com/kubeedge/kubeedge.git
- Run make all BUILD_WITH_CONTAINER=false
- Observe the Go linker error due to the emojis and warning text being passed to -ldflags.

**Anything else we need to know?**:
I have already fixed this locally by redirecting the warnings in hack/lib/golang.sh to >&2

**Environment**:
- Kubernetes version (use `kubectl version`): N/A (Build-time error)
- KubeEdge version(e.g. `cloudcore --version` and `edgecore --version`): master branch
- <details><summary>Cloud nodes Environment:</summary>

--- a/hack/lib/golang.sh
+++ b/hack/lib/golang.sh
@@ -59,10 +59,10 @@ kubeedge::version::get_version_info() {
 
   GIT_VERSION=$(git describe --tags --abbrev=14 "${GIT_COMMIT}^{commit}" 2>/dev/null)
   if [[ -z "${GIT_VERSION}" ]]; then
-    echo 
-    echo "⚠️  No git tags found."
-    echo "⚠️  Falling back to default version: v0.0.0"
-    echo "⚠️  To avoid this, consider running: git fetch --tags"
+    echo >&2
+    echo "⚠️  No git tags found." >&2
+    echo "⚠️  Falling back to default version: v0.0.0" >&2
+    echo "⚠️  To avoid this, consider running: git fetch --tags" >&2
 
     GIT_VERSION="v0.0.0"
   fi

